### PR TITLE
SLVS-1530 Fix use shared binding when shared binding exists and solution is standalone

### PR DESCRIPTION
**Pull request by vnaskos-sonar**
[SLVS-1530](https://sonarsource.atlassian.net/browse/SLVS-1530)

Regression fix

Expectations:
When shared binding exists for a solution and the solution is in standalone mode:
1. The "Bind" button on the shared binding goldbar notification should open the connected mode window and bind
2. On the Connected mode window, the "Use shared binding" should be visible, and clicking it should bind

[SLVS-1530]: https://sonarsource.atlassian.net/browse/SLVS-1530?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/ConnectedMode/UI/ManageBinding/ManageBindingViewModel.cs [hunk 1]
```diff
@@ -138,6 +138,13 @@ public async Task InitializeDataAsync()
         var displayBindStatus = new TaskToPerformParams<AdapterResponse>(DisplayBindStatusAsync, UiResources.FetchingBindingStatusText,
                 UiResources.FetchingBindingStatusFailedText) { AfterProgressUpdated = OnProgressUpdated};
         await ProgressReporter.ExecuteTaskWithProgressAsync(displayBindStatus);
+
+        if (!IsCurrentProjectBound)
+        {
+            var detectSharedBinding = new TaskToPerformParams<AdapterResponse>(CheckForSharedBindingAsync, UiResources.CheckingForSharedBindingText,
+                UiResources.CheckingForSharedBindingFailedText) { AfterProgressUpdated = OnProgressUpdated };
```
Comment (vnaskos-sonar) [inline src/ConnectedMode/UI/ManageBinding/ManageBindingViewModel.cs:145]: Not sure if the `UiResources.CheckingForSharedBindingFailedText` is ever displayed, I couldn't test or figure out how to trigger it.

File: src/ConnectedMode/UI/Resources/UiResources.resx [hunk 2]
```diff
@@ -357,6 +357,12 @@
   <data name="FetchingBindingStatusText" xml:space="preserve">
     <value>Fetching binding status...</value>
   </data>
+  <data name="CheckingForSharedBindingFailedText" xml:space="preserve">
+    <value>Checking for shared configuration...</value>
```
Comment (gabriela-trutan-sonarsource) [inline src/ConnectedMode/UI/Resources/UiResources.resx:361]: The resources should be swapped. You put the checking for shared configuration in the resource with the "Failed" key.

File: src/ConnectedMode/UI/ManageBinding/ManageBindingViewModel.cs (hunk 1 above)
Comment (gabriela-trutan-sonarsource) [reply src/ConnectedMode/UI/ManageBinding/ManageBindingViewModel.cs:145]: It is shown when an error occurs during the TaskToPerform is executed. So if you throw an exception in the CheckForSharedBindingAsync, you can see it appears: 
image

**Review by gabriela-trutan-sonarsource — CHANGES_REQUESTED** [review]
A minor resource swap

File: src/ConnectedMode/UI/ManageBinding/ManageBindingViewModel.cs (hunk 1 above)
Comment (gabriela-trutan-sonarsource) [reply src/ConnectedMode/UI/ManageBinding/ManageBindingViewModel.cs:145]: It is shown when an error occurs during the TaskToPerform is executed. So if you throw an exception in the CheckForSharedBindingAsync, you can see it appears:

image

**Review by gabriela-trutan-sonarsource — APPROVED** [review]

**Pull request merged**